Orange/widgets/unsupervised/owpca.py

```diff
@@ -292,33 +290,18 @@ def commit(self):
                 # Compute the full transform (MAX_COMPONENTS components) once.
                 self._transformed = self._pca(self.data)
             transformed = self._transformed
-            d = self.data.domain
-
-            names = [var.name for var in chain(d.attributes, d.class_vars, d.metas) if var]
-            proposed = [var.name for var in transformed.domain.attributes[:self.ncomponents]]
-            unique = get_unique_names(names, proposed)
-
-            transformed_attributes = []
-            for var, u in zip(transformed.domain.attributes[:self.ncomponents], unique):
-                if var.name != u:
-                    transformed_attributes.append(var.copy(u))
-                else:
-                    transformed_attributes.append(var)
 
             domain = Domain(
-                transformed_attributes,
+                transformed.domain.attributes[:self.ncomponents],
                 self.data.domain.class_vars,
                 self.data.domain.metas
             )
             transformed = transformed.from_table(domain, transformed)
             # prevent caching new features by defining compute_value
-            comp_names = [a.name for a in self._pca.orig_domain.attributes]
-            meta_var = get_unique_names(comp_names, 'component')
-
             dom = Domain(
                 [ContinuousVariable(a.name, compute_value=lambda _: None)
                  for a in self._pca.orig_domain.attributes],
-                metas=[StringVariable(name=meta_var)])
+                metas=[StringVariable(name='component')])
```

Comment:
I fear you reverted too much.

Outputs "Data" and "Transformed data" don't need checking (as far as I see) but the third output, Components, still does. If apply PCA to data that contains a variable named "component", I get a table with two variables named "component".